File: js알고리즘/20220914/1_정답.py
def GCD(a, b):
    if b == 0: return a
    else: return GCD(b, a%b)

def solution(A):
    A.sort(reverse=True)

    retVal = GCD(A[0], A[1])


    for i in range(2, len(A)):
        retVal = GCD(retVal, A[i])

    return retVal

File: javascript-algorithm/20220914/1_정답.js
function GCD(a, b)
{
  if (b == 0)
  {
    return a;
  }
  else
  {
    return GCD(b, a % b);
  }
}

function solution(A)
{
  A.sort()
  result = GCD(A[0], A[1]);

  for (var i = 2; i < A.length; i++)
  {
    result = GCD(result, A[i]);
  }

  return result;
}

console.log(solution([15,5,50]))